src/cargo/core/shell.rs

```diff
@@ -423,6 +454,63 @@ impl ShellOut {
         Ok(())
     }
 
+    /// Prints out a message with a status to stdout. Output the specified color and bold for the query content.
+    fn message_stdout(
+        &mut self,
+        name_no_query: Vec<&str>,
+        desc_no_query: Vec<&str>,
+        query: &str,
+        color: Color,
+        justified: bool,
+    ) -> CargoResult<()> {
+        match *self {
+            ShellOut::Stream { ref mut stdout, .. } => {
+                let mut count_name = 0;
+                for message in name_no_query.iter() {
+                    count_name += 1;
+                    stdout.reset()?;
+                    write!(stdout, "{}", message)?;
+                    stdout.set_color(ColorSpec::new().set_bold(true).set_fg(Some(color)))?;
+                    if count_name != name_no_query.len() {
+                        write!(stdout, "{}", query)?;
+                    }
+                }
+
+                if !desc_no_query.is_empty() {
+                    let mut count_desc = 0;
+                    for message in desc_no_query.iter() {
```

Comment:
This looks to be the same loop as above, so can the code be refactored to avoid this duplication?

Reply:
thank you，refactored the code.